src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy

```diff
@@ -234,4 +236,17 @@ class SurrealPersistenceService implements PersistenceService {
         return result
     }
 
+    /**
+     * Surreal allows 16KB as maximum size for a payload
+     * This method will truncate the payload if it exceeds 14 KB
+     * Truncate the payload if it exceeds the maximum size
+     */
+    protected static truncateLargeCondaFile(String condafile) {
+        int maxSize = 14 * 1024 //14 KB for the file and 2 KB for the rest of the fields
```

Comment:
let's make a config option for this as usual

Reply:
done